lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -50,6 +50,9 @@ def compute(self, student, course_structure, scores_client, submissions_scores):
         ):
             self._compute_block_score(student, descendant_key, course_structure, scores_client, submissions_scores)
 
+        # self.scores may hold outdated data, force it to refresh on next access
+        lazy.invalidate(self, 'scores')
```

Comment:
It turns out that, for the problem being tested in `test_create` below, the `scores` property returns `(0, 1)`, which was exactly the value I was mocking, so I never noticed this bug.

We need to invalidate this property in `compute`, as `self.locations_to_weighted_scores` has been updated.